code/Model/SiteTree.php

```diff
@@ -2711,6 +2712,43 @@ public function getPageLevel() {
 	}
 
 	/**
+	 * @param string $action
+	 * @return Controller
+	 */
+	public function getAssociatedController($action = null) {
+		$controllerName = $this->getControllerName($action);
+		return Injector::inst()->create($controllerName, $this);
+	}
+
+	/**
+	 * @param string $action
+	 * @return string
+	 */
+	public function getControllerName($action = null) {
+		$action = ucfirst($action); // Needed as Injector service names are case sensitive
```

Comment:
It’s how `ModelAsController::controller_for()` works currently. I added the comment inline to describe why I *think* it’s like that...

I’ve just spotted a mistake I made when trying to tidy this logic up... `Page_Controller_Action` should take priority over `Page_Controller`. Will fix.